babel/board.js

```diff
@@ -104,11 +138,16 @@ export class Board {
     }
 
     needsShuffle() {
-
+        return !this.hasMatch();
     }
 
     hasMatch() {
-        return !this.needsShuffle();
+        let chunks = iterchunks(this.orbs);
+        // [[[1, 2, 3], [2, 3, 4]], [[3, 4, 5], [4, 5, 6]]] becomes
+        //  [[1, 2, 3], [2, 3, 4], [3, 4, 5], [4, 5, 6]]
+        let flatChunks = _.flattenDepth(chunks, 1);
+        let hasWideStyleMatch = _.some(_.map(flatChunks, hasMatchInSingleRow));
+        return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
```

Comment:
This is why all my tests put the possible matches in the bottom-right hand corner of the board. I wanted it to match towards the end, to see what would happen.